src/examples/signal_generator.cpp

```diff
@@ -0,0 +1,139 @@
+/**
+ * @file signal_generator.cpp
+ * @author Luciana Moreno Borges (youlumoreno.borges@gmail.com)
+ * @brief Configure the components, connect them and start the state machine
+ * @version 0.1
+ * @date 2019-01-25
+ * 
+ * @copyright Copyright (c) 2019
+ * 
+ */
+#include <memory>
+#include <QApplication>
+#include <QTimer>
+#include <hyro/core/StateMachine.h>
+#include <hyro/utils/DynamicPropertyAccess.h>
+#include <hyro/widgets/WidgetCollection.h>
+#include <hyro/utils/StateMachineSpinner.h>
+#include <hyro/SignalGeneratorComponent.h>
+#include <hyro/DigitalConverterComponent.h>
+
+using namespace hyro;
+/**
+ * @brief Main loop
+ * 
+ * @param argc 
+ * @param argv 
+ * @return int 
+ */
+int main(int argc, char **argv)
+{
+  hyro::LogConfig config;
+  config.level = hyro::LogLevel::CRITICAL;
+  
+  /*Components configuration */
+  hyro::HyroLoggerManager::Configure(config);
+    StateMachine signal_generator_sm(std::make_shared<SignalGeneratorComponent>("/signal_generator"_uri));
+    StateMachine digital_converter_sm(std::make_shared<DigitalConverterComponent>("/digital_converter"_uri));
+    ComponentConfiguration signal_generator_conf = ComponentConfiguration(
+       "{"
+        "outputs: {"
+           "signals: { protocol: 'api'},"
+           "fix_dynamic: {protocol: 'api'}"
+        "}" 
+      "}"
+
+    );
+
+    ComponentConfiguration digital_converter_conf = ComponentConfiguration(
+    "{"
+      "inputs: {"
+        "signals: { protocol: 'api' }"
+      "},"
+      "outputs: {"
+        "digital_signals: { protocol: 'api'}"
+      "}"
+    "}"
+    );
+
+    auto remote_endpoint = ConnectionConfiguration("{ signals: { endpoint: '/signal_generator/signals'} }");
+    
+    /*State transistions */
+
+     /*Initialize the component with configurations */
+    signal_generator_sm.init(signal_generator_conf);
+    signal_generator_sm.start();
+    signal_generator_sm.connect();
+    signal_generator_sm.check();
+
+     /*Initialize the component with configurations */
+    digital_converter_sm.init(digital_converter_conf);
+    digital_converter_sm.start();
+    digital_converter_sm.connect(remote_endpoint);
+    digital_converter_sm.check();
+
+    CancellationTokenSource cancellation_token;
+
+    /*Create spinners*/
+    StateMachineSpinner signal_generator_spinner(signal_generator_sm, cancellation_token,10ms);
+    StateMachineSpinner digital_converter_spinner(digital_converter_sm, cancellation_token,10ms);
+    
+    /* Initialize dynamic properties with a default value */
+    float m_amplitude = 0.0f;
```

Comment:
Here you don't need include the prefix **"m_"** in the variables. We just include **"m_"** in the member class variables.